### PR TITLE
Feat/payment step type

### PR DESCRIPTION
**Pull request by BeBlood**

### CODE REVIEW

File: .docker/build/php/php.ini [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+date.timezone = 'Europe/Paris';
+short_open_tag = Off;
+magic_quotes_gpc = Off;
+register_globals = Off;
+session.auto_start = Off;
+error_reporting = E_ALL;
+log_errors = On;
+display_errors = On;
+error_log = /dev/stderr;
+memory_limit = -1;
```
Comment (konandrum) [inline .docker/build/php/php.ini]: To fix : 128M

File: Command/CleanTransactionCommand.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    {
+        parent::__construct();
+    }
+
+    protected function configure()
+    {
+        $this
+            ->setName('app:transaction:clean')
+            ->setDescription('Clean old transaction that stay in pending or created status after a given delay (in hours)')
+            ->setHelp('Clean transaction that stay in pending or created status after a given delay (in hours)')
```
Comment (konandrum) [inline Command/CleanTransactionCommand.php]: Change use P1D for exemple

File: Command/CleanTransactionCommand.php [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        }
+
+        $qb = $om->createQueryBuilder();
+
+        $qb
+            ->select('t')
+            ->from(Transaction::class, 't')
+            ->where('t.updatedAt < :updatedAt')
+            ->andWhere('t.status = :statusCreated OR t.status = :statusPending')
+            ->setParameter('updatedAt', (new \DateTime('now'))->sub(new \DateInterval(sprintf('PT%sH', $hours))))
```
Comment (konandrum) [inline Command/CleanTransactionCommand.php]: Change, use the given argument

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+# Utils
+
+vendor: composer.json composer.lock
+	make -C $$PWD composer-install
+
+.PHONY: composer-add-github-token
+composer-add-github-token:
+	docker-compose run --rm $(target_container) composer config --global github-oauth.github.com $(token)
+
+.PHONY: composer-update
```
Comment (konandrum) [inline Makefile]: .PHONY: composer-update
composer-update:
	docker-compose run --rm $(target_container) php -d memory_limit=-1 /usr/local/bin/composer update $(options)

File: Makefile [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+composer-update:
+	docker-compose run --rm $(target_container) composer update
+
+.PHONY: composer-install
```
Comment (konandrum) [inline Makefile]: .PHONY: composer-install
composer-install:
	docker-compose run --rm $(target_container) composer install $(options)

File: Makefile [hunk 6]
```diff
@@ -0,0 +1,51 @@
+# Variables
+
+target_container ?= php
```
Comment (konandrum) [inline Makefile]: Really used ?

File: Resources/config/config.yml [hunk 7]
```diff
@@ -1,2 +1,12 @@
 imports:
     - { resource: step_types.yml }
+    - { resource: event_actions.yml }
```
Comment (konandrum) [inline Resources/config/config.yml:3]: To include if we need IDCIStepBundle => In the documentation

File: Step/Type/PaymentStepType.php [hunk 8]
```diff
@@ -66,24 +74,27 @@ public function configureOptions(OptionsResolver $resolver)
...
+                'currency_code' => null,
                 'callback_url' => null,
                 'customer_email' => null,
                 'customer_id' => null,
                 'description' => null,
                 'return_url' => null,
             ])
             ->setNormalizer('amount', function (Options $options, $value) {
-                return intval($value, 10);
+                return null !== $value ? intval($value, 10) : -1;
```
Comment (konandrum) [inline Step/Type/PaymentStepType.php]: keep null value

File: Step/Type/PaymentStepType.php [hunk 9]
```diff
@@ -66,24 +74,27 @@ public function configureOptions(OptionsResolver $resolver)
...
+            })
+            ->setNormalizer('item_id', function (Options $options, $value) {
```
Comment (konandrum) [inline Step/Type/PaymentStepType.php]: Keep it to null

File: Step/Type/PaymentStepType.php (hunk 8 above)
Comment (BeBlood) [reply Step/Type/PaymentStepType.php]: Fixed in 88f667bc281e2a0b047e19831679f7e71c35441f

File: Step/Type/PaymentStepType.php (hunk 9 above)
Comment (BeBlood) [reply Step/Type/PaymentStepType.php]: Fixed in 88f667bc281e2a0b047e19831679f7e71c35441f

File: Makefile (hunk 6 above)
Comment (BeBlood) [reply Makefile]: Fixed in 27d41f822f32d2b15ce1dfd1f214c1e81c5ee876

File: Makefile (hunk 5 above)
Comment (BeBlood) [reply Makefile]: Fixed in 27d41f822f32d2b15ce1dfd1f214c1e81c5ee876

File: Makefile (hunk 4 above)
Comment (BeBlood) [reply Makefile]: Fixed in 27d41f822f32d2b15ce1dfd1f214c1e81c5ee876

File: .docker/build/php/php.ini (hunk 1 above)
Comment (BeBlood) [reply .docker/build/php/php.ini]: Fixed in 27d41f822f32d2b15ce1dfd1f214c1e81c5ee876

File: Command/CleanTransactionCommand.php (hunk 3 above)
Comment (BeBlood) [reply Command/CleanTransactionCommand.php]: Fixed in 7fa6997f3ebe3e6a72ec9dd9350cb28b780da9a5

File: Command/CleanTransactionCommand.php (hunk 2 above)
Comment (BeBlood) [reply Command/CleanTransactionCommand.php]: Fixed in 7fa6997f3ebe3e6a72ec9dd9350cb28b780da9a5

**Pull request merged**